dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/CompletionStageRxInvokerImpl.java

```diff
@@ -29,19 +29,24 @@
 import javax.ws.rs.core.Response;
 //Liberty code change start
 import org.eclipse.microprofile.context.ManagedExecutor;
-import org.eclipse.microprofile.context.ThreadContext;
 //Liberty code change end
 
 public class CompletionStageRxInvokerImpl implements CompletionStageRxInvoker {
     private WebClient wc;
     private ExecutorService ex;    
   //Liberty code change start  
-    private boolean isManagedExecutor = false;    
+    private boolean isManagedExecutor = false;
+    private boolean isExecutorNull = false;
+    private ExecutorService defaultManagedExecutor = null;
     CompletionStageRxInvokerImpl(WebClient wc, ExecutorService ex) {
         this.ex = ex;        
         if (this.ex instanceof ManagedExecutor) {            
             isManagedExecutor = true;
         }
+        if (this.ex == null) {
+            isExecutorNull = true;
+            defaultManagedExecutor = ManagedExecutor.builder().propagated().build();
```

Comment:
This is definitely an improvement on creating less ManagedExecutor instances, but I suspect there is still a memory leak here, albeit a lesser one.  Am I right in assuming that a new instance of CompletionStageRxInvokerImpl would be created with each clientbuilder.build()?  If so, every time that is invoked, then once the user is done with the client, it will leak one ManagedExecutor.  These could build up over time.  It looks like javax.ws.rs.client.Client has a `close` method, where you could shut down the executor instance if one is created for it, and that would prevent the memory leak.  However, that would have some other implications, which might or might not be desirable - specifically that it will attempt to cancel/interrupt ongoing async operations on the executor and prevent further async operations from being requested.  I should probably chat with Andy about what sort of behavior he wants to see for the default (no executor specified) path, both with respect to this as well as with regard to context clearing (mentioned in other comments).